### ISSUE TITLE
Single TV show in multiple locations shows duplicate in TV shows page

### ISSUE THREAD

**Issue by brianjmurrell** [jellyfin/jellyfin#904]
**Describe the bug**
If I have multiple directories in a library and in those directories I have different episodes of the same TV show, the TV show overview screen will show duplicates of that same show.

**To Reproduce**
1. Add multiple directories to a library
1. In each of those directories, create a directory for a given show you have episodes of
1. Distribute the episodes to each the show directories in the library
1. Go to the TV show overview screen

**Expected behavior**
I should see just one show icon which I can click on and will see all of the episodes from the various library directories.

**System (please complete the following information):**
 - OS: Fedora 29
 - Browser: Chrome
 - Jellyfin Version: 10.1.0

**Comment by rrado** [comment]
Same issue here on all browsers and mobile 
- OS: Ubuntu 18.04.2
- Jellyfin Version: 10.3.3

This situation can happen if you have for example : 
- 1 folder for English version
- 1 folder for original language version


This issue was not present on Emby (tested with Emby 4.1.1.0 & 4.2.0.5 beta)
Since this issue is present since months, I'm wondering if it would not be better to go back to Emby ?

**Major consequences of this bug : **
Let's consider I'm currently watching an "original language" version episode.
- once each episode is finished : when clicking on "next episode", jellyfin plays the same episode ! (but the English version)
- continue watching does not show the correct episode : same issue as above

**Comment by rrado** [comment]
One additional information
This feature has been implemented in Emby 3.6.0.51 version (02 October 2018) : https://emby.media/community/index.php?/topic/63528-36051-multi-version-episodes/

I don't know if this part of the code was still shared between Emby and Jellyfin ?

**Comment by anthonylavado** [comment]
@rrado - No, Jellyfin was based on Emby 3.5.2. Anything after that was closed off.

**Comment by brianjmurrell** [comment]
Still an issue.

**Comment by parkdalton** [comment]
Still issue.

**Comment by anthonylavado** [comment]
Can anyone give me an exact directory breakdown to try and replicate this?

**Comment by brianjmurrell** [comment]
I was going to but I don't seem to have this problem any more.

**Comment by bnichs5** [comment]
@anthonylavado I'm still having the issue in the latest Jellyfin. Looks like this.    

"\tv\scooby doo where are you!\season 01\S01E01\" Then I have four different versions to the same episode in that folder and each come up as a different item.

**Comment by juicetin** [comment]
Echoing @bnichs5 's comment above.

I have:

"/tv/some show name/Season 3/some show name (2005) s01e01 some episode name/some show name (2005) s01e01 some episode name - 1080p.mp4"
"/tv/some show name/Season 3/some show name (2005) s01e01 some episode name/some show name (2005) s01e01 some episode name - 720p.mp4"
"/tv/some show name/Season 3/some show name (2005) s01e01 some episode name/some show name (2005) s01e01 some episode name - 480p.mp4"
"/tv/some show name/Season 3/some show name (2005) s01e01 some episode name/some show name (2005) s01e01 some episode name - 360p.mp4"

But each of these come up as a separate entry in the show list.

**Comment by dkanada** [comment]
Jellyfin uses brackets to check for multi-version files. Something like `Filename [Version]` should work I think.

**Comment by juicetin** [comment]
Thanks for the tip @dkanada.

I gave that a go:

"/tv/some show name/Season 3/some show name (2005) s01e01 some episode name/some show name (2005) s01e01 some episode name [1080p].mp4"
"/tv/some show name/Season 3/some show name (2005) s01e01 some episode name/some show name (2005) s01e01 some episode name [720p].mp4"
"/tv/some show name/Season 3/some show name (2005) s01e01 some episode name/some show name (2005) s01e01 some episode name [480p].mp4"
"/tv/some show name/Season 3/some show name (2005) s01e01 some episode name/some show name (2005) s01e01 some episode name [360p].mp4"

But unfortunately they still get picked up as separate episodes.

Would you happen to have this working and have some example full paths to the working files? I wonder if there's something in the full paths above that are messing with Jellyfin's version detection.

**Comment by giejay** [comment]
Same issue for me for 1080p versions vs 720p in same season folder.

**Comment by JustAMan** [comment]
I feel this should be a feature request, as I don't think JF ever supported that (thus it cannot be classified as "bug").

**Comment by brianjmurrell** [comment]
IMHO, bugs are bugs.  Whether they were introduced since the Emby fork or inherited from Emby is orthogonal.

Displaying duplicate shows is a bug (again, IMHO) no matter how you slice it.

**Comment by jcvara** [comment]
Same issue here with different seasons in different folders. I ran out of space in one HDD so I had to split a show, first folder has season 1 and 2, second folder has the rest of them:

/data/series/The Flash (2014)/01x01 - City of Heroes.mkv
...

/data/series_tmp/The Flash (2014)/03x01 - Flashpoint.mkv
...

Everything shows correctly, except I have two entries in the main page, each one with the corresponding seasons showing inside, mirroring the disposition in the filesystem.

Jellyfin info (showing on dashboard):

Version: 10.4.3
Operating System: Linux
Architecture: X64

I'm using the linuxserver/jellyfin:latest docker image in a QNAP nas.

**Comment by jdinkel** [comment]
Same issue here with the multiple versions of episodes.  "Series_Name - S01E01 - H265.mp4" and "Series_Name - S01E01 - VP9.webm" show up like two different episodes in JellyFin.

**Comment by brianjmurrell** [comment]
@jdinkel I would posit that what you describe is quite a bit different than what is being described in this ticket and that solving your problem should not be part of solving this ticket.

This ticket deals with completely non-overlapping duplication that can be de-duplicated without any loss or choice having to be made.  Your's is clearly a case of overlapping duplication for which a different solution is needed.

I would suggest opening a new ticket for your issue.

**Comment by holow29** [comment]
Unsure about having files in multiple folders (which is what this issue was about according to the author), but for multiple versions of media inside the same folder, look at https://github.com/jellyfin/jellyfin/issues/1996

**Comment by ArabCoders** [comment]
i have the exact same problem

folder structure 
/media/tv/show name/season 03/show name - SXEX - title.mkv
/media/tv_archive/show name/season 01/show name - SXEX - title.mkv
/media/tv_archive/show name/season 02/show name - SXEX - title.mkv

will be presented as separate entities in the listing. one for S3 and another one for S1-2

**Comment by dscott1** [comment]
> i have the exact same problem
> 
> folder structure
> /media/tv/show name/season 03/show name - SXEX - title.mkv
> /media/tv_archive/show name/season 01/show name - SXEX - title.mkv
> /media/tv_archive/show name/season 02/show name - SXEX - title.mkv
> 
> ## will be presented as separate entities in the listing. one for S3 and another one for S1-2
> nevermind, the setting was under advanced and i somehow missed it. if someone wondering it's under
> 
> admin > libraries > manage library > show advanced settings > Automatically merge series that are spread across multiple folders.
> 
> it was off by default.

This solved my issue. Thanks ArabCoders. I'm surprised this option isn't enabled by default

**Comment by brianjmurrell** [comment]
> admin > libraries > manage library > show advanced settings > Automatically merge series that are spread across multiple 
folders.

I do already have that set.

**Comment by sarunas-zilinskas** [comment]
No such option is available when library content is selected as "mixed content". 

Which is sad.

**Comment by Xyz00777** [comment]
i hope im right here
I have a similar problem, i had a tvshow in one folder (than i reorganized my whole tvshow folder structure) and now i have every episode of these show (happend only at one show) in jellyfin 2 times with different folder locations. The old location (not exist anymore) and the new location. clearly it cant play half of the episodes because the old location dont exist anymore but it dont remove these episodes. its really anoying

**Comment by sarunas-zilinskas** [comment]
> i hope im right here
> I have a similar problem, i had a tvshow in one folder (than i reorganized my whole tvshow folder structure) and now i have every episode of these show (happend only at one show) in jellyfin 2 times with different folder locations. The old location (not exist anymore) and the new location. clearly it cant play half of the episodes because the old location dont exist anymore but it dont remove these episodes. its really anoying

Juat re-scan the library. It does not have anything to do with this issue.

**Comment by akashkj** [comment]
I believe this issue is still present.

**Comment by vpsj** [comment]
Just got Jellyfin ( After getting fed up of Plex finally) and this is still an issue. I have even checked the "Automatically merge series that are spread across multiple folders." option, but the problem remains no matter how much I try to scan and rescan the library and refresh the metadata.

**Comment by HornheaDD** [comment]
This is evidently still going on. See link for image: https://imgur.com/mNORciA.jpg

18 of 19 seasons are in one location, the most recent season is in another. Like vpsj, I've selected "merge series" and its still happening.

**Comment by Karlosll91** [comment]
Happening to me too, I had enabled the ‘Automatically merge series that are spread across multiple folders’ since the beginning. Can you guys please check if the issue is happening only when the folders are in 2 diferent volumes? in my case I am using Linux and have 2 season in Hard disk 1 and the 3rd season in Hard Disk 2

**Comment by racemaniac** [comment]
Also adding a comment hoping this gets fixed soon :).

**Comment by andriuhaynes** [comment]
Same issue here

**Comment by dpieski** [comment]
> Same issue here, on linuxserver latest docker container

Can you provide an example of the directory structure of a season of a particular series in one location and a different season of the series in a different location? 

I'd be interested in trying to figure this out and I am trying to recreate the issue.

**Comment by exterrestris** [comment]
My general directory structure is as follows:
```
/media/shows/rips/series-name/
/media/shows/downloads/series-name/
```
The library is set up with `/media/shows/rips` and `/media/shows/downloads` as the two root directories. Any show that I only own part of on DVD or Blu-ray will have the relevant disc rips in the `rips` directory, with anything provided by Sonarr etc. in the `downloads` directory. I don't use season directories in my library as they seem superfluous, but I don't see that mattering for this issue - the point is that a given series can be split across multiple library roots

**Comment by d03j** [comment]
same here.

running 10.10.5 in a docker.io/jellyfin/jellyfin:latest container.

I use mergerfs with my media drives but point jelyfin to the underlying filesystems as per [mergerfs's recommendation](https://trapexit.github.io/mergerfs/faq/limit_drive_spinup/#what-if-you-only-query-already-active-drives), with my several /TV.Shows folders mapping to /media/TV.Shows/3, /media/TV.Shows/4 and /media/TV.Shows/5 in the container.

Currently I have S01 and S02 with 6 episodes each spread through drives 3 and 5:

'/media/TV.Shows/3/show.name.[tmdbid-xxx]/S01'
'/media/TV.Shows/5/show.name.[tmdbid-xxx]/S02'

I have "Automatically merge series that are spread across multiple folders" ticked in my Shows library.

backdrop.jpg, folder.jpg, landscape.jpg, logo.png, and tvshow.nfo are present in both folders and seem to have been created at the same time as each folder.

the show appears twice in the recently added section (both at home and under tv shows) but not in jellyfin.example.com/web/#/tv.html?topParentId=xxxxxxxx&collectionType=tvshows

**Comment by LhymeFruit** [comment]
Still unsolved? not even syslinks work.

**Comment by azhirov** [comment]
Still issue.

```
/data/shows/Fallout.S02.1080p.AMZN.WEB-DL.H.264
/data/shows/Fallout (Season 1) BDRemux 1080p
```

<img width="654" height="462" alt="Image" src="https://github.com/user-attachments/assets/1f75a187-541a-4d17-953f-4f41454b3c9d" />

**Comment by zotabee** [comment]
> Still issue.
> 
> ```
> /data/shows/Fallout.S02.1080p.AMZN.WEB-DL.H.264
> /data/shows/Fallout (Season 1) BDRemux 1080p
> ```
> 
> <img alt="Image" width="654" height="462" src="https://private-user-images.githubusercontent.com/29061184/538916260-1f75a187-541a-4d17-953f-4f41454b3c9d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjkxNDk4MjEsIm5iZiI6MTc2OTE0OTUyMSwicGF0aCI6Ii8yOTA2MTE4NC81Mzg5MTYyNjAtMWY3NWExODctNTQxYS00ZDE3LTk1M2YtNGY0MTQ1NGIzYzlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTIzVDA2MjUyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4ZWJhOWM2MDBkOTJlOGY4NTg5ODRmOGQ2ZWE5YmMxOWY1MzgxMjQyMGYxZDM2ZTdkODcwN2JlY2Q0NTE4OTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Fox28rQyS8IaE-Df1e3cRBdUy93Y1FwiWbmdHAVTYws">

 

@azhirov Read this : https://jellyfin.org/docs/general/server/media/shows/

**Comment by brianjmurrell** [comment]
> Read this : https://jellyfin.org/docs/general/server/media/shows/

What exactly in that long page are you referring to?

Are you trying to use the example storage layouts to explain that we should not have multiple storage locations that have the same show split across them?  Because if so, that's simply not always practical.  Jellyfin supports multiple storage locations.  It should handle merging the contents of them into a single cohesive library without visually displaying any separations of data like it were duplicates.

**Comment by dpieski** [comment]
Pretty sure that they are trying to tell you that your folder naming does not follow the correct convention. (1) You do not have a series folder; and (2) Your season folders are incorrectly named.

Your example locations are not multiple storage locations. You're just showing random seasons all thrown together in one "shows" folder without identifying what the show actually is.

An example of multiple storage locations that this issue is about would be:
```
/data/media/tv/Series/Season 01/
/share/media/tv/Series/Season 02/
```

**Comment by AnotherGithubRando** [comment]
Still having this issue on 10.11.7. This happens with series spread across multiple drives, and with when episodes appear in multiple directories on a single drive, or a combination of the two.

**Comment by d03j** [comment]
Still happening

<img width="425" height="343" alt="Image" src="https://github.com/user-attachments/assets/07619a34-83b3-40a0-afa9-d1c061f1e0a2" />

folder structure:
```
/drive1/TV.Shows/ZeroZeroZero.[tmdbid-92137]
/drive2/TV.Shows/ZeroZeroZero.[tmdbid-92137]
```


Server version: 10.11.7
Web version: 10.11.7
Build version: 10.11.7

docker image: docker.io/jellyfin/jellyfin:latest

**Comment by FritzHeiden** [comment]
Still occuring in 10.11.8. You can use mergerFS to create a merged mount point from multiple directories as a workaround.

**Comment by d03j** [comment]
> Still occuring in 10.11.8. You can use mergerFS to create a merged mount point from multiple directories as a workaround.

You can, but it isn't recommended. MergerFS [says](https://trapexit.github.io/mergerfs/latest/faq/limit_drive_spinup) when "setting up software like Plex, Jellyfin, etc. point to the underlying filesystems. Not mergerfs".

**Comment by NiklasTheis** [comment]
I was having this issue in version 10.11.11. I updated to 12.0-rc3 and the TV-shows are being merged and displayed as expected. Multiple drives are merged to one tile.
